frontend/src/plugin/registry.tsx

```diff
@@ -16,6 +17,17 @@ interface SectionFuncProps {
 
 export type sectionFunc = (resource: KubeObject) => SectionFuncProps | null | undefined;
 
+export type getAppLogoFunc = (props: {
+  logoType?: LOGO_TYPE;
+  theme?: string;
+  [key: string]: any;
+}) => JSX.Element | null;
+
+export enum LOGO_TYPE {
+  'SMALL' = 'SMALL',
+  'LARGE' = 'LARGE',
```

Comment:
This looks a bit odd. Maybe we can make it be just be an object? (so it doesn't have to be used from Ts)

@illume , do you have a suggestion for this?

Reply:
It can be written a bit more clearly like this: `"small" | "large"`

If we pass in a component instead this would turn into the props interface. The function type here seems to already use props.